Documentation/ceph-block.md

```diff
@@ -142,6 +142,150 @@ kubectl delete -n rook-ceph cephblockpools.ceph.rook.io replicapool
+PVC created earlier.
+
+```console
+kubectl create -f cluster/examples/kubernetes/ceph/csi/rbd/snapshot.yaml
+```
+
+### Verify RBD Snapshot Creation
+
+```console
+$ kubectl get volumesnapshotclass
+NAME                      DRIVER             DELETIONPOLICY   AGE
+csi-rbdplugin-snapclass   rook-ceph.rbd.csi.ceph.com   Delete           3h55m
+
+$ kubectl get volumesnapshot
+NAME               READYTOUSE   SOURCEPVC   SOURCESNAPSHOTCONTENT   RESTORESIZE   SNAPSHOTCLASS             SNAPSHOTCONTENT                                    CREATIONTIME   AGE
+rbd-pvc-snapshot   true         rbd-pvc                             1Gi           csi-rbdplugin-snapclass   snapcontent-79090db0-7c66-4b18-bf4a-634772c7cac7   3h50m          3h51m
+```
+
+The snapshot will be ready to restore to a new PVC when `READYTOUSE` field of the
+volumesnapshot is set to true
+
+### Restore the snapshot to a new PVC
+
+In
+[pvc-restore](/cluster/examples/kubernetes/ceph/csi/rbd/pvc-restore.yaml),
+`dataSource` should be the name of the `VolumeSnapshot` previously
+created. The kind should be the `VolumeSnapshot`.
+
+Create a new PVC from the snapshot
+
+```console
+kubectl create -f cluster/examples/kubernetes/ceph/csi/rbd/pvc-restore.yaml
+```
+
+### Verify RBD Clone PVC Creation
+
+```bash
+$ kubectl get pvc
+NAME              STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS           AGE
+rbd-pvc           Bound    pvc-84294e34-577a-11e9-b34f-525400581048   1Gi        RWO            rook-ceph-block        34m
+rbd-pvc-restore   Bound    pvc-575537bf-577f-11e9-b34f-525400581048   1Gi        RWO            rook-ceph-block        8s
+```
+
+## RBD snapshot resource Cleanup
+
+To clean your cluster of the resources created by this example, run the following:
+
+if you have tested snapshot, delete snapshotclass, snapshot and pvc-restore
+created to test snapshot feature
+
+```console
+kubectl delete -f cluster/examples/kubernetes/ceph/csi/rbd/pvc-restore.yaml
+kubectl delete -f cluster/examples/kubernetes/ceph/csi/rbd/snapshot.yaml
+kubectl delete -f cluster/examples/kubernetes/ceph/csi/rbd/snapshotclass.yaml
+```
+
+## RBD Volume Cloning
+
+The CSI Volume Cloning feature adds support for specifying existing PVCs in the
+dataSource field to indicate a user would like to clone a Volume.
+
+A Clone is defined as a duplicate of an existing Kubernetes Volume that can be
+consumed as any standard Volume would be. The only difference is that upon
+provisioning, rather than creating a "new" empty Volume, the back end device
+creates an exact duplicate of the specified Volume.
+
+Refer
+[clone-doc](https://kubernetes.io/docs/concepts/storage/volume-pvc-datasource/)
+for more info.
+
+### Volume Clone Prerequisites
+
+ Requires kubernetes v1.16+ which supports volume clone.
+
+### Volume Cloning
+
+In
+[pvc-clone](/cluster/examples/kubernetes/ceph/csi/rbd/pvc-clone.yaml),
```

Comment:
To link to the yamls, you see other examples near the top of this doc where it links with the full url based on the branch name. There are a few instances of these links in this doc to update.
```
(https://github.com/rook/rook/tree/{{ branchName }}/cluster/examples/kubernetes/ceph/csi/rbd)
```